### PR TITLE
c# 7 refractoring

### PR DESCRIPTION
**Pull request by dansiegel**
general code cleanup with inline variable declarations

### CODE REVIEW

**Comment by dvorn** [comment]
@dansiegel Seemingly the build server was not upgraded to c# 7.  Until it is you may use Roslyn compiler as NuGet package: https://www.nuget.org/packages/Microsoft.Net.Compilers/.

File: Source/Xamarin/Prism.Forms/Common/PageUtilities.cs [hunk 1]
```diff
@@ -45,29 +42,26 @@ public static void DestroyPage(Page page)
 
         private static void DestroyChildren(Page page)
         {
-            if (page is MasterDetailPage)
+            if (page is MasterDetailPage mdp)
```
Comment (bartlannoeye) [inline Source/Xamarin/Prism.Forms/Common/PageUtilities.cs]: do we want abbreviations?

File: Source/Xamarin/Prism.Forms/Common/PageUtilities.cs [hunk 2]
```diff
@@ -165,12 +153,12 @@ public static Page GetOnNavigatedToTargetFromChild(Page target)
         {
             Page child = null;
 
-            if (target is MasterDetailPage)
-                child = ((MasterDetailPage)target).Detail;
-            else if (target is TabbedPage)
-                child = ((TabbedPage)target).CurrentPage;
-            else if (target is CarouselPage)
-                child = ((CarouselPage)target).CurrentPage;
+            if (target is MasterDetailPage mdp)
```
Comment (bartlannoeye) [inline Source/Xamarin/Prism.Forms/Common/PageUtilities.cs]: idem

File: Source/Xamarin/Prism.Forms/Mvvm/ViewModelLocator.cs [hunk 3]
```diff
@@ -35,8 +35,7 @@ public static void SetAutowireViewModel(BindableObject bindable, bool? value)
 
         private static void OnAutowireViewModelChanged(BindableObject bindable, object oldValue, object newValue)
         {
-            bool? bNewValue = (bool?)newValue;
-            if (bNewValue.HasValue && bNewValue.Value)
+            if (newValue is bool shouldAutowire && shouldAutowire)
```
Comment (bartlannoeye) [inline Source/Xamarin/Prism.Forms/Mvvm/ViewModelLocator.cs]: I might not be used yet to the C# 7 syntax, but I think the old syntax is easier to read/understand than this.

**Comment by dansiegel** [comment]
@bartlannoeye I think it may be a combination of you're not used to it, and preference. I actually think it's easier to follow with inline declarations. That said as @dvorn has pointed out the build server doesn't seem to be upgraded yet, so it wouldn't be very good to merge this until AppVeyor supports C# 7

**Comment by brianlagunas** [comment]
@dansiegel I wish you would have talked to me about this first.  This really isn't a pressing issue or a real benefit to the framework.  We might be able to talk about this later as C# 7 becomes more popular and prevalent in the dev space, and after AppVeyor is compatible with C# 7.  For now, this isn't something that "moves the needle" if you will.

**Pull request closed without merging**